Automattic/simplenote-ios
# Add auto versioning for betas and hotfixes

This PR adds the lanes for beta and hotfix versioning.
The PR builds on top of https://github.com/Automattic/simplenote-ios/pull/355 and should be merged after that one. 

# To Test
Usually, this lane is used on the current release branch, but, since it will commit and push, we'll checkout a new test branch and use an optional parameter to override the standard behaviour.

1. Checkout a new `release/xx.xx` branch and push it with `-u`. Ex: `release/4.9`. 
2. Run `bundle exec fastlane new_beta_release base_version:4.9`, or the version you checked out before. 
3. At the prompt, verify that the lane reads the correct current version (`4.8.0.1`) and confirm.
4. Verify that the script completes without errors.
5. Verify that the version has been updated to `4.8.0.2` and the files committed and pushed. 
6. Verity that a `4.8.0.2` tag has been added and pushed. 

### Clean up
7. Delete the `4.8.0.2` tag from your local repository and from GitHub.
8. Delete the `release/4.9` branch from your local repository and from GitHub.
